packages/mobile/src/components/share-drawer/ShareDrawer.tsx

```diff
@@ -134,35 +193,110 @@ export const ShareDrawer = () => {
       )
       const audioStartOffsetConfig =
         content.track.duration && content.track.duration >= 20 ? '-ss 10 ' : ''
-      const session = await FFmpegKit.execute(
-        `${audioStartOffsetConfig}-i ${streamMp3Url} -filter_complex "gradients=s=1080x1920:c0=000000:c1=434343:x0=0:x1=0:y0=0:y1=1920:duration=10:speed=0.0225:rate=60[bg];[0:a]aformat=channel_layouts=mono,showwaves=mode=cline:n=1:s=1080x200:scale=cbrt:colors=#ffffff[fg];[bg][fg]overlay=format=auto:x=0:y=1275" -pix_fmt yuv420p -vb 50M -t 10 ${storyVideoPath}`
-      )
-      // TODO(nkang): Add loading state
-      const returnCode = await session.getReturnCode()
 
-      if (ReturnCode.isSuccess(returnCode)) {
+      let stickerUri: string | undefined // Have to capture the sticker image first because it doesn't work if you get the dominant colors first (mysterious).
+
+      try {
+        stickerUri = await captureStickerImage()
+      } catch (e) {
+        reportToSentry({
+          level: ErrorLevel.Error,
+          error: new Error(e),
+          name: 'Share to IG Story error - generate sticker step'
+        })
+        toast({ content: messages.shareToStoryError, type: 'error' })
+        return
+      }
+      if (!stickerUri) {
+        reportToSentry({
+          level: ErrorLevel.Error,
+          error: new Error('Sticker screenshot unsuccessful'),
+          name: 'Share to IG Story error - generate sticker step (sticker undefined)'
+        })
+        toast({ content: messages.shareToStoryError, type: 'error' })
+        return
+      }
+      let dominantColorsResult: Color[]
+      let dominantColorHex1: string
+      let dominantColorHex2: string
+      if (trackImageUri) {
+        try {
+          dominantColorsResult = await getDominantColors(trackImageUri)
+        } catch (e) {
+          reportToSentry({
```

Comment:
thinking this report + toast could maybe be combined into a helper fn?

Reply:
yeah for sure it got really repetitive haha